src/main/java/com/eternalcode/parcellockers/locker/controller/LockerPlaceController.java

```diff
@@ -17,26 +15,49 @@
 import org.bukkit.event.block.BlockPlaceEvent;
 import org.bukkit.inventory.ItemStack;
 import org.bukkit.inventory.PlayerInventory;
+import org.bukkit.inventory.meta.ItemMeta;
 import org.bukkit.plugin.Plugin;
 
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
 import java.util.UUID;
 
 public class LockerPlaceController implements Listener {
 
     private final PluginConfiguration config;
-    private final MiniMessage miniMessage;
     private final Plugin plugin;
-    private final LockerRepositoryImpl databaseService;
+    private final LockerRepository databaseService;
     private final NotificationAnnouncer announcer;
+    private final Map<UUID, Boolean> lockerCreators = new HashMap<>();
 
-    public LockerPlaceController(PluginConfiguration config, MiniMessage miniMessage, Plugin plugin, LockerRepositoryImpl databaseService, NotificationAnnouncer announcer) {
+    public LockerPlaceController(PluginConfiguration config, Plugin plugin, LockerRepository databaseService, NotificationAnnouncer announcer) {
         this.config = config;
-        this.miniMessage = miniMessage;
         this.plugin = plugin;
         this.databaseService = databaseService;
         this.announcer = announcer;
     }
 
+    private static boolean compareMeta(ItemStack first, ItemStack second) {
+        ItemMeta firstMeta = first.getItemMeta();
+        if (firstMeta == null) {
+            return false;
+        }
+
+        ItemMeta secondMeta = second.getItemMeta();
+
+        if (secondMeta == null) {
+            return false;
+        }
+
+        if (first.getType() != second.getType()) {
+            return false;
+        }
+
+        return new HashSet<>(firstMeta.getLore()).containsAll(secondMeta.getLore())
```

Comment:
Isn't it better to just use the compareTo method at this point?

Reply:
`equals` method is more intuitive and directly checks for equality.
`compareTo` is generally used for ordering and might be less clear in this context.

Reply:
No. The current way you compare the name and lore of items creates a new hashset each time, which can be inefficient. The compareTo method compares the items directly (with enchants etc) which is a better solution. Let the rest of the members comment on this.